scripts/migrate-attachments.pl

```diff
@@ -110,33 +144,80 @@
   print "Attachments migrated: $migrated\n";
 }
 
+if ($options{class}) {
+  if ($options{class}->[0] eq $options{class}->[1]) {
+    die "Source and destination must be different\n";
+  }
+  my ($source, $dest) = @{$options{class}};
+
+  my ($total) = $dbh->selectrow_array('
+    SELECT COUNT(*) 
+      FROM attachments 
+           JOIN attachment_storage_class ON attachments.attach_id = attachment_storage_class.id 
+     WHERE attachment_storage_class.storage_class = ?', undef, $source);
+
+  confirm(sprintf 'Update %d attachments class from %s to %s?',
+    $total, @{$options{class}});
+
+  my $sth = $dbh->prepare('
+      SELECT attach_id 
+        FROM attachments 
+             JOIN attachment_storage_class ON attachments.attach_id = attachment_storage_class.id 
+       WHERE attachment_storage_class.storage_class = ? ORDER BY attach_id DESC');
+  $sth->execute($source);
+  my ($count, $updated) = (0, 0);
+  while (my ($attach_id) = $sth->fetchrow_array()) {
+    indicate_progress({total => $total, current => ++$count});
+
+    my $attachment = Bugzilla::Attachment->new({id => $attach_id, cached => 1});
+
+    # Update the class of the attachment
+    $attachment->current_storage($dest)->set_class();
+
+    $updated++;
+  }
+  print "\n";
+  print "Attachments updated: $updated\n";
```

Comment:
Looks like `$count` and `$updated` are the same thing here? We can remove one of them.

Reply:
$count is the number of rows currently process from the total rows returned from the database. $updated, $deleted, $migrated, are the number of rows that actually had some action taken with the row. So they need to be different values.

Reply:
In this case, `$updated` and `$count` will always be the same value when they are accessed.

Reply:
> In this case, `$updated` and `$count` will always be the same value when they are accessed.

Ah I see what you mean. Yes for the 'class' section, since there is no conditional in the while loop, then the two values are always the same anyway. Well rather than push a new change and since it is no harm, I will leave it for consistency :)